src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -426,7 +426,11 @@ partition_downloader::download_log_with_capped_size(
     size_t total_size = 0;
     for (auto it = offset_map.rbegin(); it != offset_map.rend(); it++) {
         const auto& meta = it->second;
-        if (total_size + meta.size_bytes >= max_size) {
+        if (
+          !staged_downloads.empty()
+          && total_size + meta.size_bytes >= max_size) {
+            // At the moment we can't safely restore the partition without
+            // downloading at least one segment.
```

Comment:
Are the limitations there the ones I asked about earlier with regards to delta offsets? Or something else.